### ISSUE TITLE
Curious "zero or negative width"-warning for barcharts

### ISSUE THREAD

**Issue by dahilzen** [mhkeller/layercake#67]
First of all thanks for this fantastic framework!

I think I found a little bug in the barchart-example. As soon as padding.left is 100 or higher this error occurs:

`[LayerCake] Target div has zero or negative width. Did you forget to set an explicit width in CSS on the container?
`

the chart appears anyway. Then i found out that the error does not occur as long as padding.left + padding.right < 100. 
I have no idea how to prevent this error... Can somebody reproduce this error?

**Comment by mhkeller** [comment]
Thanks for letting me know. That's just a little warning to help people who may have forgotten to set a width and height. I *think* it's appearing because before the div is measured, the default width and height of a graphic is 100x100. So with padding greater than that, it will cause a warning – but that will no longer be the case once the page fully loads. I should be able to suppress this by making this warning being dependent on the environment. That is to say, if we are in non-ssr mode and there's window, then ignore this. I'll have to test what the best solution is for ssr.

**Comment by mhkeller** [comment]
@dahilzen is it causing any other problems for you or is it just that there's a console error?

**Comment by mhkeller** [comment]
It looks like even only checking for a width client-side will still cause the error because, for some reason, that function is fired before the `containerWidth` and `containerHeight` are updated for some reasons. Going to look into this more...

**Comment by dahilzen** [comment]
There is another error, that the width of the bars is negative. You can work around this by Math.abs the width of the rect in the Bars-Component.

Apart from these errors, everything works perfectly. The chart looks exactly as it should.

**Comment by mhkeller** [comment]
Can you post an example in the REPL showing these negative bars?

**Comment by dahilzen** [comment]
Actually, I can't. The error does not appear in the REPL. Seems to be a problem with my environment.  Sorry about that.

**Comment by mhkeller** [comment]
Could you post a screenshot?

**Comment by dahilzen** [comment]
![image](https://user-images.githubusercontent.com/16721203/154569468-9212d743-64d8-418d-a085-ce38df3a45e7.png)
![image](https://user-images.githubusercontent.com/16721203/154568753-1b3d331e-2f4a-4faf-987d-78084f920ee0.png)

padding.top = 0
padding.right = 30
padding.bottom = 0
padding.left = 150

and i changed some stuff in the Bar.svelte

```
<script>
  import { getContext } from "svelte";
  import { deUmlaut } from "./umlaute";
  const { data, xGet, yGet} = getContext("LayerCake");

  export let fill;
  export let yKey;
  export let xKey;
  export let suffix;
  export let format;
  export let labelThreshold;

</script>

<g class="___bar-group">
  {#each $data as d, i}
    <rect
      class="___bar_rects ___bar_rect_{deUmlaut(d[yKey])}"
      data-id={i}
      x={0}
      y={$yGet(d)}
      height={25}
      width={$xGet(d)}
      {fill}
    />
    {#if $xGet(d) < labelThreshold}
      <text
        class="___chart_label"
        text-anchor="start"
        x={$xGet(d) + 3}
        y={$yGet(d)}
        dy="17.5">{format(d[xKey])} {" " + suffix}</text
      >
    {:else}
      <text
        class="___chart_label ___chart_label_{deUmlaut(d[yKey])} ___chart_label_inner"
        text-anchor="end"
        x={$xGet(d) - 3}
        y={$yGet(d)}
        dy="17.5">{format(d[xKey])} {" " + suffix}</text
      >
    {/if}
  {/each}
</g>
```

**Comment by mhkeller** [comment]
Can you log out what you're getting for these values

```js
console.log('width', $width);
console.log('xScale.domain()', $xScale.domain());
console.log('xScale.range()', $xScale.range());
console.log('xDomain', $xDomain);
console.log('xRange', $xRange);
```

**Comment by mhkeller** [comment]
I've [added a check](https://github.com/mhkeller/layercake/pull/68/files#diff-c14b959d6ebbaaee817ef4b3e8bf97ca252b46fd6e7f3575c2db3ddc7b8cabc0R263-R267) that should suppress this warning until the component is mounted and the container width is properly measured. You could try installing this branch directly from GitHub to see if that fixes the issue with `npm install --save mhkeller/layercake#reactivity`.

**Comment by dahilzen** [comment]
It works! And the rect-width-error is also gone. Thank you!

**Comment by mhkeller** [comment]
This should be fixed in 6.1.0 https://github.com/mhkeller/layercake/blob/main/CHANGELOG.md#610
